backend/user_service/controllers/userController.js

```diff
@@ -0,0 +1,55 @@
+const { db, auth } = require("../config/firebaseConfig.js");
+const admin = require('firebase-admin');
+
+// Function to create a new user and add them to the Firestore collection
+const addToUserCollection = async (req, res) => {
+    const { uid, email, displayName } = req.body;
```

Comment:
Perhaps username could be a better fitting variable name rather than displayname, but its fine if you have no time.

Reply:
This is changed in add-username!